satpy/readers/smos_l2_wind.py

```diff
@@ -0,0 +1,173 @@
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""SMOS L2 wind Reader.
+
+Data can be found here after register:
+https://www.smosstorm.org/Data2/SMOS-NRT-wind-Products-access
+Format documentation at the same site after register:
+SMOS_WIND_DS_PDD_20191107_signed.pdf
+"""
+
+import logging
+import numpy as np
+from datetime import datetime
+from pyresample.geometry import AreaDefinition
+from satpy.readers.netcdf_utils import NetCDF4FileHandler, netCDF4
+
+logger = logging.getLogger(__name__)
+
+
+class SMOSL2WINDFileHandler(NetCDF4FileHandler):
+    """File handler for SMOS L2 wind netCDF files."""
+
+    @property
+    def start_time(self):
+        """Get start time."""
+        return datetime.strptime(self['/attr/time_coverage_start'], "%Y-%m-%dT%H:%M:%S Z")
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        return datetime.strptime(self['/attr/time_coverage_end'], "%Y-%m-%dT%H:%M:%S Z")
+
+    @property
+    def platform_shortname(self):
+        """Get platform shortname."""
+        return self.filename_info['platform_shortname']
+
+    @property
+    def platform_name(self):
+        """Get platform."""
+        return self['/attr/platform']
+
+    def get_metadata(self, data, ds_info):
+        """Get metadata."""
+        metadata = {}
+        metadata.update(data.attrs)
+        metadata.update(ds_info)
+        metadata.update({
+            'platform_shortname': self.platform_shortname,
+            'platform_name': self.platform_name,
+            'sensor': self['/attr/instrument'],
+            'start_time': self.start_time,
+            'end_time': self.end_time,
+            'level': self['/attr/processing_level'],
+        })
+
+        return metadata
+
+    def available_datasets(self, configured_datasets=None):
+        """Automatically determine datasets provided by this file."""
+        logger.debug("Available_datasets begin...")
+
+        # This is where we dynamically add new datasets
+        # We will sift through all groups and variables, looking for data matching
+        # the geolocation bounds
+        handled_variables = set()
+
+        # Iterate over dataset contents
+        for var_name, val in self.file_content.items():
+            # Only evaluate variables
+            if not isinstance(val, netCDF4.Variable):
+                continue
+            if (var_name in handled_variables):
+                logger.debug("Already handled, skipping: %s", var_name)
+                continue
+            handled_variables.add(var_name)
+            new_info = {
+                'name': var_name,
+                'file_type': self.filetype_info['file_type'],
+            }
+            yield True, new_info
+
+    def _mask_dataset(self, data):
+        """Mask out fill values ( and remove the _FillValue from attributes)."""
+        fill = data.attrs.pop('_FillValue')
+        return data.where(data != fill)
```

Comment:
You can leave `_FillValue` in the attrs, but just replace it with `nan` (I suppose this is floats ?)

Reply:
ok